File: educational-system-consumer-admin-9000/src/main/java/edu/hubu/consumeradmin/service/IAnnouncementTypeService.java
package edu.hubu.consumeradmin.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.admin.AnnouncementType;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "admin-provider-service")
public interface IAnnouncementTypeService {

    @GetMapping("/announcementType/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/announcementType/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "announcementTypeId") String orderBy);

    @GetMapping("/announcementType/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "announcementTypeId") String orderBy
    );

    @PostMapping("/announcementType/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "announcementTypeId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.AnnouncementType announcementType
    );

    @PostMapping("/announcementType/")
    CommonResult save(@RequestBody AnnouncementType announcementType);

    @PutMapping("/announcementType/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody AnnouncementType announcementType);

    @DeleteMapping("/announcementType/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/announcementType/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/announcementType/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/admin/IAnnouncementTypeService.kt
package edu.hubu.commons.service.admin

import edu.hubu.commons.model.admin.AnnouncementType;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IAnnouncementTypeService {
    fun save(t: AnnouncementType): AnnouncementType
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: AnnouncementType): AnnouncementType
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): AnnouncementType
    fun selectAllOrderBy(order: String = "announcementTypeId"): List<AnnouncementType>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "announcementTypeId"): List<AnnouncementType>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "announcementTypeId", t: edu.hubu.commons.mybatis.model.AnnouncementType): List<edu.hubu.commons.mybatis.model.AnnouncementType>
}
